src/connectors/bus.cpp

```diff
@@ -24,19 +24,23 @@ along with Fritzing.  If not, see <http://www.gnu.org/licenses/>.
 #include "connectoritem.h"
 #include "../model/modelpart.h"
 
-Bus::Bus(BusShared * busShared, ModelPart * modelPart) : QObject()
+Bus::Bus(BusShared * busShared, ModelPart * modelPart) : QObject(),
+    m_connectors(),
+    m_subConnector(nullptr),
+    m_busShared(busShared),
+    m_modelPart(modelPart)
 {
-	m_busShared = busShared;
-	m_modelPart = modelPart;
 }
 
-const QString & Bus::id() const {
-	if (m_busShared == NULL) return ___emptyString___;
-
```

Comment:
I was getting massively confused on how this code was working so I used the syntax we use at work and my own projects. I have no issue conforming to your coding style. I would actually suggest using c++17's std::optional in the future instead of empty string. I'll fix that as well.

Reply:
Probably we can not switch to C++17 that soon. But yes, there are to many nullptr checks in the Fritzing code, some of them looking quite defensive.

Reply:
Using std::optional would require some heavy refactoring. I would not recommend it in the short term anyway. C++17 is wonderful but having access to C++14 is so welcome :).